src/l500/l500-serializable.cpp

```diff
@@ -34,11 +35,46 @@ namespace librealsense
     {
         json j = json::parse(json_content);
 
+        // Set of options that should not be set in the loop
+        std::set< rs2_option > options_to_ignore{ RS2_OPTION_SENSOR_MODE };
+
+        // We have to set the sensor mode (resolution) first
+        auto & sensor_mode = _depth_sensor.get_option( RS2_OPTION_SENSOR_MODE );
+        auto found_sensor_mode = j.find( get_string( RS2_OPTION_SENSOR_MODE ) );
+        if( found_sensor_mode != j.end() )
+        {
+            float sensor_mode_val = found_sensor_mode.value();
+            sensor_mode.set( sensor_mode_val );
+        }
+
+        // If a non custom preset is used, we should ignore all the settings that are automatically
+        // set by the preset
+        auto found_iterator = j.find( get_string( RS2_OPTION_VISUAL_PRESET ) );
+        if( found_iterator != j.end() )
+        {
+            auto found_preset = rs2_l500_visual_preset( int( found_iterator.value() ));
```

Comment:
A direct conversion from value to the enum raised an exception, I had to convert to int first.